#This notebook generates PCA plots of the protome/pisa data and the RNA-seq data

#open up files
```{r}

#library(rstudioapi)
library(stringr)
library(ggplot2)
library(dplyr)
library(ggforce)
# Using spreadsheet generated by PISA analysis.Rmd. 
pisa_prot<- read.csv("Inputs/PISA_global_changes.csv")

pisa_prot<- na.omit(pisa_prot)
```

#prepare data frame to do a pca analysis
  columns are proteins
  rows are samples

```{r}

pisa_mat<- pisa_prot[,c(1:17)]

pisa_mat<- pisa_mat[!duplicated(pisa_mat$Accession),]

rownames(pisa_mat)<- pisa_mat[,1]

pisa_mat<- pisa_mat[,-c(1)]

cols.num<- colnames(pisa_mat)

pisa_mat[cols.num]<- sapply(pisa_mat[cols.num], as.numeric)

pisa_mat<- na.omit(pisa_mat)

pisa_mat_t<- t(pisa_mat)

```
```{r}

pisa_pca<- prcomp(pisa_mat_t)

```


Plotting the pca
```{r}

plot(pisa_pca$x[,1], pisa_pca$x[,2])

```
making a scree plot
```{r}
pisa.pca.var<- pisa_pca$sdev^2

pisa.pca.var<- round(pisa.pca.var/sum(pisa.pca.var)*100,1)

```


Using ggplot to make a nice pca plot
```{r}
pisa.pca_data<- data.frame(sample= rownames(pisa_pca$x),
                         x=pisa_pca$x[,1],
                         y = pisa_pca$x[,2])

pisa.pca_data$cell<-c(rep("Control Global",4) , 
                      rep("Control PISA",4),
                      rep("DZNep Global", 4),
                      rep ("DZNep PISA",4))


```
#PC1 vs PC2
```{r}

a<-ggplot(data=pisa.pca_data,aes(label=sample,x=x,y=y, color = cell)) + 
 
  ggtitle("Proteome PCA Plot") +

  geom_point(aes( size = 4)) +
  
  scale_color_manual(values = c("Control Global"= "#18288d", 
                                "Control PISA"="#a2a9d1",
                                "DZNep Global"="#877817",
                                "DZNep PISA"="#c3bb8b"))+

  xlab(paste("PC1: ",pisa.pca.var[1],"%",sep="")) +

  ylab(paste("PC2: ",pisa.pca.var[2],"%",sep=""))+
  theme_classic()+
  theme(legend.text = element_text(size = 10))+
  guides(color=guide_legend(override.aes = list(size = 5)),
         shape = guide_legend(override.aes = list(size=5)))
 
a

#ggsave("PISA_PCA PC1.eps", a, device= "eps", width = 6, height = 3)
```

# looking at what contributes to PC1
```{r}

#getting the loading scores

loadingScores<- pisa_pca$rotation[,1]

#getting the magnitudes of loading scores

geneScores<- abs(loadingScores)

#soring from high to low

geneScoreRanked<- sort(geneScores, decreasing = T)

#getting top 5 format 3

top5genes<- names(geneScoreRanked[1:10])

top_pisa_pca<-pisa_pca$rotation[top5genes,1]
top_pisa_pca

```

# PCA plot of the RNAseq results
```{r}
#preparing the df
rna_seq_2<- read.csv("Inputs/rlog_transformed_counts.csv")

mat_rna<- rna_seq_2

rownames(mat_rna)<- mat_rna[,1]

mat_rna<- mat_rna[,-c(1)]

cols.num<- colnames(mat_rna)

mat_rna[cols.num]<- sapply(mat_rna[cols.num], as.numeric)

mat_rna<- na.omit(mat_rna)

mat_rna<- t(mat_rna)

rna_pca<- prcomp(mat_rna)



```

# Plotting the pca
```{r}

plot(rna_pca$x[,1], rna_pca$x[,2])

```
making a scree plot
```{r}
rna_pca.var<- rna_pca$sdev^2

rna_pca.var<- round(rna_pca.var/sum(rna_pca.var)*100,1)

```
```{r}
barplot(rna_pca.var, main="Scree Plot", xlab="Principal Component", ylab="Percent Variation")

```

Using ggplot to make a nice pca plot
```{r}
rna_pca_data<- data.frame(sample= rownames(rna_pca$x),
                         x=rna_pca$x[,1],
                         y = rna_pca$x[,2])

rna_pca_data$cell<-c(rep(c("Control","DZNep"),times =4))


```
#PC1 vs PC2
```{r}
library(ggforce)
a<-ggplot(data=rna_pca_data,aes(label=sample,x=x,y=y, color = cell)) + 
 
  ggtitle("Transcriptome PCA") +

  geom_point(aes( size = 4)) +
  
  scale_color_manual(values = c("Control"= "#18288d", "DZNep"="#877817"))+

  xlab(paste("PC1: ",rna_pca.var[1],"%",sep="")) +

  ylab(paste("PC2: ",rna_pca.var[2],"%",sep=""))+
   geom_mark_ellipse(aes(color = as.factor(cell)), expand = unit(2, "mm"), show.legend = F )+
  #geom_encircle(aes(group = type), color = "red", s_shape = 0.01, expand = 0.04)+
  theme_classic()+
  theme(legend.text = element_text(size = 10))+
  guides(color=guide_legend(override.aes = list(size = 5)),
         shape = guide_legend(override.aes = list(size=5)))
 
a

ggsave("rna_global_PCA PC1.eps", a, device= "eps", width = 6, height = 3)
```

## looking at what contributes the most to PC1
```{r}

#getting the loading scores

loadingScores<- rna_pca$rotation[,1]

#getting the magnitudes of loading scores

geneScores<- abs(loadingScores)

#soring from high to low

geneScoreRanked<- sort(geneScores, decreasing = T)

#getting top 5 format 3

top5genes<- names(geneScoreRanked[1:10])

top_bc_pca<-rna_pca$rotation[top5genes,1]
top_bc_pca

```

